### PR TITLE
re-ordering and organizing on day 2 - discuss

### PR DESCRIPTION
**Pull request by ChristinaLK**
Holy changes, Batman!  

* reordering material on Day 2 (teaching practices/motivation in morning, carpentries, lessons in afternoon)
* re-ordering and adding text to teaching practices to flow into motivation
* re-ordering and adding text to lesson objectives lesson, removed some text
* moving information about SWC/DC lessons to various other places (somewhat re-creating/modifying #245)
* adding some info re: DC lessons

I did this all as (mostly) separate commits, so if we like one set of modifications but not another, I might be able to cherry-pick the desired changes onto the current gh-pages version. :D

### CODE REVIEW

File: _episodes/15-practices.md [hunk 1]
```diff
@@ -227,6 +184,67 @@ Peak rule.
     While it has been criticized for not strongly predicting what's remembered,
     it's always worth trying to end a lesson on a high note.
 
+Instructor guides.
+:   Each Software and Data Carpentry lesson has an instructor's guide, with notes
```
Comment (ErinBecker) [inline _episodes/15-practices.md]: Several of the DC lessons don't yet have instructors' guides. The ones that do have the guides linked from our main lessons page.

Comment (ChristinaLK) [inline _episodes/15-practices.md]: I wasn't sure - will change...

**Comment by gvwilson** [comment]
Can you please summarize the final order after these changes?  (I'm finding it hard to track through the diff.)

**Comment by ChristinaLK** [comment]
morn:  live coding / break / carpentry teaching practices / motivation
afternoon: carpentry overview / break / lessons (both objective writing + the SWC/DC lessons)

Note that I split some of the material we had about the lessons: the "carpentries" episode now has an overview of the lesson topics + how to contribute (sort of) and the "lessons" episode talks first about learning objectives + reverse instructional design generally, and then more specifically about the general learning objectives of the SWC/DC lessons.

**Comment by ChristinaLK** [comment]
Also, there's still some casual text in there - I will try to update today.

File: _episodes/15-practices.md [hunk 2]
```diff
@@ -228,6 +185,71 @@ Peak rule.
...
     it's always worth trying to end a lesson on a high note.
 
+Instructor guides.
+:   Many of the Software and Data Carpentry lessons have an 
+    instructor's guide, with notes 
+    from instructors who have already taught the material.  This can be a valuable 
+    resource when preparing lessons, especially when teaching a lesson for the first time.  
+    The Software Carpentry instructor guides are linked on each lesson page; the 
+    instructor guides for Data Carpentry lessons are on 
+    their [main lesson page][here]({{ site.dc_site }}/lessons/).
```
Comment (lexnederbragt) [inline _episodes/15-practices.md]: Two times text in square brackets, I suspect it should be 

`main lesson page [here]({{ site.dc_site }}/lessons/).`

Also, not all lessons listed on the DC lessons page have links to the instructor guides...

File: _episodes/19-carpentries.md [hunk 3]
```diff
@@ -146,6 +146,42 @@ There are also materials in development and testing for:
 
 Other Data Carpentry lessons are in the incubator stage.
 
+## Existing Lessons
```
Comment (lexnederbragt) [inline _episodes/19-carpentries.md]: This duplicates the links to lessons, check the text just above this location.

File: _episodes/21-lessons.md [hunk 4]
```diff
@@ -434,22 +341,111 @@ but as Scott pointed out in
...
+how to build modular programs out of small functions
+that can be read, tested, and re-used.
+However,
+these concepts turn out to be hard to convey to people
+who are still learning the syntax of a programming language
+(forest and trees),
+so in practice the programming lessons focus primarily on
+the mechanics of doing common operations in those languages.
+
+### Data Carpentry
```
Comment (lexnederbragt) [inline _episodes/21-lessons.md]: I wold remove hyperlinks to lesson material from the Data Carpentry section below.

**Comment by lexnederbragt** [comment]
Looks good.

One thing I saw that we currently already have multiple places with links to the main lessons (*current* 15-carpentries and 20-lessons). The move proposed in the PR (into 19-carpentries) actually places another copy under the existing one.

This PR's overview in 21-lessons does not have links for SWC lessons, but it does for DC lessons - seems redundant.

**Comment by ChristinaLK** [comment]
okay, should have addressed Lex's comments, including the duplicated lesson links.  This PR got kind of out of hand, but I think it should be in good shape now.  @gvwilson can you merge?

**Pull request merged**